unit.go

```diff
@@ -0,0 +1,338 @@
+// Copyright ©2013 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package unit
+
+import "fmt"
+
+// Unit is a type a value with generic SI units. Most useful for
+// translating between dimensions, for example, by multiplying
+// an acceleration with a mass to get a force. Please see the
+// package documentation for further explanation.
+type Unit struct {
+	dimensions Dimensions // Map for custom dimensions
+	formatted  string
+	value      float64
+}
+
+// New creates a new variable of type Unit which has the value
+// specified by value and the dimensions specified by the
+// base units struct. The value is always in SI Units.
+//
+// Example: To create an acceleration of 3 m/s^2, one could do
+// myvar := CreateUnit(3.0, &Dimensions{unit.LengthDim: 1, unit.TimeDim: -2})
+func New(value float64, d Dimensions) *Unit {
```

Comment:
I surmise this is a leftover from the `Uniter` interface but: why the `*Unit` indirection?
wouldn't returning it by value put less pressure on the GC?